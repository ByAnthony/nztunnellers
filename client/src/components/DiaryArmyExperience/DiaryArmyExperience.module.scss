@use 'src/index.scss';

.war-service {
  @include index.featured-button-link;

  flex: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: index.$xs-margin 0;
  padding: index.$s-margin;
  font-size: 1.25rem;
  color: index.$secondary-color;

  p {
    color: index.$secondary-color;
  }

  @include index.desktop {
    padding: index.$padding-card;
    flex: 50%;
  }

  @include index.large-desktop {
    padding: index.$padding-card;
    flex: 50%;
  }
}
