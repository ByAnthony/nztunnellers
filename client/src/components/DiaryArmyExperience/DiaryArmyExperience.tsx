import { ArmyExperience } from '../../types/tunneller';
import STYLES from '../ProfileDiary/ProfileDiary.module.scss';
import STYLES_WWI from './DiaryArmyExperience.module.scss';

type props = {
  tunnellerId: number,
  armyExperience: ArmyExperience[],
}

export function ArmyExperienceList({ militaryExperience }:
  {militaryExperience: ArmyExperience[] | []}) {
  if (militaryExperience.length > 0) {
    return (
      <>
        {militaryExperience.map((experience) => {
          const displayDurationAndCountry = () => {
            const isUk = (country: string) => (country === 'United Kingdom' ? `the ${country}` : country);
            if (experience.duration !== null && experience.country !== null) {
              return <p>{`${experience.duration} in ${isUk(experience.country)}`}</p>;
            }
            if (experience.duration !== null && experience.country === null) {
              return <p>{experience.duration}</p>;
            }
            if (experience.duration === null && experience.country !== null) {
              return <p>{experience.country}</p>;
            }
            return null;
          };

          if (experience.unit !== 'Other' && experience.conflict !== null) {
            return (
              <li className={STYLES['fullwidth-secondary-card']} key={experience.unit}>
                <p>{experience.conflict}</p>
                <span>{ experience.unit }</span>
              </li>
            );
          }

          if (experience.unit === 'Other') {
            if (experience.conflict !== null && experience.duration === null) {
              return (
                <li className={STYLES['fullwidth-secondary-card']} key={experience.unit}>
                  <span>{experience.conflict}</span>
                </li>
              );
            }
            return (
              <li className={STYLES['fullwidth-secondary-card']} key={experience.unit}>
                <span>{experience.conflict}</span>
                <p>{experience.duration}</p>
              </li>
            );
          }
          return (
            <li className={STYLES['fullwidth-secondary-card']} key={experience.unit}>
              <span>{ experience.unit }</span>
              { displayDurationAndCountry() }
            </li>
          );
        })}
      </>
    );
  }
  return null;
}

export function DiaryArmyExperience({ tunnellerId, armyExperience }: props) {
  return (
    <>
      <h2>Army Experience</h2>
      <ArmyExperienceList militaryExperience={armyExperience} />
      <a href={`/tunnellers/${tunnellerId}/wwi-timeline`} className={STYLES_WWI['war-service']} aria-label="Open the World War I timeline.">
        <div>
          <p>World War I (1914-1918)</p>
          <span>New Zealand Tunnellers</span>
        </div>
        <div className={STYLES['arrow-right']}>&rarr;</div>
      </a>
    </>
  );
}
