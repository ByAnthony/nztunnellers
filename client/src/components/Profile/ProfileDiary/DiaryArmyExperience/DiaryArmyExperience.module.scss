@use 'src/index.scss';

.line-margin {
  padding-top: index.$xxs-margin;
}

.war-service {
  flex: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: index.$xs-margin 0;
  padding: index.$s-margin;
  font-size: 1.25rem;
  color: index.$secondary-color;

  @include index.featured-button-link;

  p {
    padding-bottom: index.$xxs-margin;
    color: index.$secondary-color;
  }

  @include index.desktop {
    padding: index.$padding-card;
  }
}
