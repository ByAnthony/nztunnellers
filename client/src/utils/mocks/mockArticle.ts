import { Image } from '../../types/article';

export const mockTitle: string = 'My Awesome\\Article Title';

export const mockChapter: number = 1;

export const mockImage: Image = {
  file: 'img-123.png',
  alt: 'Accessible alt text',
};
