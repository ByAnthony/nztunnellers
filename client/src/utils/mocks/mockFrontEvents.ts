import { Date, Event, EventDetail } from '../../types/tunneller';

const mockDate = (year: string | null, dayMonth: string | null): Date => ({
  year,
  dayMonth,
});

export const mockEvent: EventDetail = {
  description: null,
  title: null,
  image: null,
};

export const mockEventCompany: EventDetail = {
  ...mockEvent,
  description: 'Something happened that day for the Company',
  title: 'The Company',
  image: 'image.jpg',
};
export const mockEventDescription: EventDetail = { ...mockEvent, description: 'Something happened that day' };
export const mockEventEnlisted: EventDetail = { ...mockEvent, description: 'Main Body', title: 'Enlisted' };
export const mockEventPosted: EventDetail = { ...mockEvent, description: 'Reinforcement', title: 'Posted' };
export const mockEventTrained: EventDetail = { ...mockEvent, description: 'Military Camp, Somewhere', title: 'Trained' };
export const mockEventTitleAndDescription: EventDetail = { ...mockEvent, description: 'Something happened', title: 'Major Event' };

export const mockEventDiedOfDisease: EventDetail = { ...mockEvent, description: 'Military Hospital', title: 'Died of disease' };
export const mockEventKilledInAction: EventDetail = { ...mockEvent, description: 'Killed on the battlefield', title: 'Killed in action' };
export const mockEventDiedOfWounds: EventDetail = { ...mockEvent, description: 'Wounded on the battlefield', title: 'Died of wounds' };
export const mockEventDiedOfAccident: EventDetail = { ...mockEvent, description: 'Fatal accident', title: 'Died of accident' };
export const mockEventBuried: EventDetail = { ...mockEvent, description: 'Cemetery, France', title: 'Buried' };
export const mockEventGrave: EventDetail = { ...mockEvent, description: '1 A 26', title: 'Grave reference' };

const mockDeathEvent = (deathEvent: EventDetail): Record<string, Event[]> => ({
  1916: [
    {
      date: mockDate('1916', '27 September'),
      event: [mockEventCompany],
    },
    {
      date: mockDate('1916', '19 November'),
      event: [mockEventDescription],
    },
  ],
  1917: [
    {
      date: mockDate('1917', '8 August'),
      event: [mockEventDescription],
    },
    {
      date: mockDate('1917', '8 August'),
      event: [deathEvent, mockEventBuried, mockEventGrave],
    },
  ],
});

export const mockFrontEventsWithDiedOfDisease:
    Record<string, Event[]> = mockDeathEvent(mockEventDiedOfDisease);

export const mockFrontEventsWithKilledInAction:
    Record<string, Event[]> = mockDeathEvent(mockEventKilledInAction);

export const mockFrontEventsWithDiedOfWounds:
    Record<string, Event[]> = mockDeathEvent(mockEventDiedOfWounds);

export const mockFrontEventsWithDiedOfAccident:
    Record<string, Event[]> = mockDeathEvent(mockEventDiedOfAccident);

export const mockFrontEvents: Record<string, Event[]> = {
  1915: [
    {
      date: mockDate('1915', '26 January'),
      event: [mockEventEnlisted],
    },
    {
      date: mockDate('1915', '30 January'),
      event: [mockEventPosted],
    },

    {
      date: mockDate('1915', '14 February'),
      event: [mockEventTrained],
    },
    {
      date: mockDate('1915', '4 May'),
      event: [mockEventTitleAndDescription],
    },
  ],
  1916: [
    {
      date: mockDate('1916', '23 June'),
      event: [mockEventDescription, mockEventDescription],
    },
    {
      date: mockDate('1916', '27 September'),
      event: [mockEventCompany],
    },
    {
      date: mockDate('1916', '19 November'),
      event: [mockEventDescription],
    },
    {
      date: mockDate('1916', '12 December'),
      event: [mockEventCompany, mockEventDescription],
    },
  ],
  1917: [
    {
      date: mockDate('1917', '9 April'),
      event: [mockEventTitleAndDescription, mockEventCompany],
    },
    {
      date: mockDate('1917', '8 August'),
      event: [mockEventDescription],
    },
  ],
  1918: [
    {
      date: mockDate('1918', '3 March April'),
      event: [mockEventTitleAndDescription],
    },
    {
      date: mockDate('1918', '13 October'),
      event: [mockEventDescription],
    },
  ],
  1919: [
    {
      date: mockDate('1919', '4 April'),
      event: [mockEventTitleAndDescription],
    },
    {
      date: mockDate('1919', '12 July'),
      event: [mockEventDescription],
    },
  ],
};
