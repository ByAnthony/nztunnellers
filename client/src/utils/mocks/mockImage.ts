import {
  Image, ImageSource, ImageNewspaper, ImageArchives, ImageBook,
} from '../../types/tunneller';

export const mockImageAucklandLibraries: string = 'https://digitalnz.org/records?text=31-B2671&tab=Images#';

export const mockImageArchives: ImageArchives = {
  location: 'Auckland War Memorial Museum',
  reference: 'MS-93/157',
};

export const mockImageFamily: string = 'Courtesy of John Doe family';

export const mockImageNewspaper: ImageNewspaper = {
  name: 'Auckland Star',
  date: '12 July 1898',
};

export const mockImageBook: ImageBook = {
  title: 'My Mock Title',
  town: 'Bethune',
  publisher: 'Publisher and Co.',
  authors: [
    {
      forename: 'Jane',
      surname: 'Doe',
    },
  ],
  year: '1925',
  page: 'p. 89',
};

export const mockImageSource: ImageSource = {
  aucklandLibraries: null,
  archives: null,
  family: null,
  newspaper: null,
  book: null,
};

export const mockImage: Image = {
  url: '1/1234.jpg',
  source: mockImageSource,
};
