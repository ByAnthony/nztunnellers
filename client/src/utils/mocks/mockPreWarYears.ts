import { ArmyExperience, Employment, PreWayYears } from '../../types/tunneller';

export const mockArmyExperience = (overrides: Partial<ArmyExperience>
    | undefined = undefined): ArmyExperience => ({
  unit: 'NZ Infantry',
  country: 'New Zealand',
  conflict: null,
  duration: '12 months',
  ...overrides,
});

export const mockEmployment = (overrides: Partial<Employment>
    | undefined = undefined): Employment => ({
  occupation: 'Goldminer',
  employer: 'Goldmining Company',
  ...overrides,
});

export const mockPreWarYears = (overrides: Partial<PreWayYears>
    | undefined = undefined): PreWayYears => ({
  armyExperience: [mockArmyExperience()],
  employment: mockEmployment(),
  residence: null,
  maritalStatus: null,
  wife: null,
  religion: null,
  ...overrides,
});
