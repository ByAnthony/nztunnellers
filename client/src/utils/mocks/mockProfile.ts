// import { Profile } from '../../types/tunneller';
// import { mockSummary } from './mockSummary';

export const mockId = 26;

// export const mockProfile = (overrides: Partial<Profile> | undefined = undefined): Profile => ({
//   id: mockId,
//   summary: mockSummary(),
//   origins: mockOrigins(),
//   preWarYears: mockPreWarYears(),
//   militaryYears: mockMilitaryYears(),
//   postServiceYears: mockPostServiceYears(),
//   sources: mockSources(),
//   image: mockImage(),
//   ...overrides,
// });
