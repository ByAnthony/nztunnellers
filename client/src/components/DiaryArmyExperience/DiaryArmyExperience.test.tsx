import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { mockId } from '../../utils/mocks/mockProfile';
import { mockArmyExperience, mockArmyExperienceList } from '../../utils/mocks/mockPreWarYears';
import { DiaryArmyExperience } from './DiaryArmyExperience';

const component = (
  <DiaryArmyExperience tunnellerId={mockId} armyExperience={mockArmyExperienceList} />
);

test('renders the component correctly', () => {
  const { asFragment } = render(component);

  expect(asFragment()).toMatchSnapshot();
});

test('renders army experience when known', () => {
  render(component);

  expect(screen.getByText('NZ Infantry')).toBeInTheDocument();
  expect(screen.getByText('12 months in New Zealand')).toBeInTheDocument();
});

test('renders army experience when duration unknown', () => {
  const componentWithoutDuration = (
    <DiaryArmyExperience
      tunnellerId={mockId}
      armyExperience={[
        {
          ...mockArmyExperience,
          duration: null,
        },
      ]}
    />
  );

  render(componentWithoutDuration);

  expect(screen.getByText('NZ Infantry')).toBeInTheDocument();
  expect(screen.getByText('New Zealand')).toBeInTheDocument();
});

test('renders army experience when country unknown', () => {
  const componentWithoutDuration = (
    <DiaryArmyExperience
      tunnellerId={mockId}
      armyExperience={[
        {
          ...mockArmyExperience,
          country: null,
        },
      ]}
    />
  );

  render(componentWithoutDuration);

  expect(screen.getByText('NZ Infantry')).toBeInTheDocument();
  expect(screen.getByText('12 months')).toBeInTheDocument();
});

test('renders army experience when country and duration unknown', () => {
  const componentWithoutDuration = (
    <DiaryArmyExperience
      tunnellerId={mockId}
      armyExperience={[
        {
          ...mockArmyExperience,
          country: null,
          duration: null,
        },
      ]}
    />
  );

  render(componentWithoutDuration);

  expect(screen.getByText('NZ Infantry')).toBeInTheDocument();
  expect(screen.queryByText('12 months')).not.toBeInTheDocument();
  expect(screen.queryByText('New Zealand')).not.toBeInTheDocument();
});

// test('renders conflict experience', () => {
//   const componentWithoutDuration = (
//     <DiaryArmyExperience
//       tunnellerId={mockId}
//       armyExperience={[mockArmyExperience({
//         unit: 'Other',
//         country: null,
//         conflict: 'South Africa War',
//         duration: '2 years',
//       })]}
//     />
//   );

//   render(componentWithoutDuration);

//   expect(screen.getByText('NZ Infantry')).toBeInTheDocument();
//   expect(screen.queryByText('12 months')).not.toBeInTheDocument();
//   expect(screen.queryByText('New Zealand')).not.toBeInTheDocument();
// });

test('does not render army experience when unknown', () => {
  render(
    <DiaryArmyExperience tunnellerId={mockId} armyExperience={[]} />,
  );

  expect(screen.queryByText('NZ Infantry')).not.toBeInTheDocument();
  expect(screen.queryByText('12 months in New Zealand')).not.toBeInTheDocument();
});
