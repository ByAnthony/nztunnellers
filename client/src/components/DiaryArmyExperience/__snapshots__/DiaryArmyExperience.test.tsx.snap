// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component correctly 1`] = `
<DocumentFragment>
  <h2>
    Army Experience
  </h2>
  <li
    class="fullwidth-secondary-card"
  >
    <span>
      NZ Infantry
    </span>
    <p
      class="line-margin"
    >
      12 months in New Zealand
    </p>
  </li>
  <a
    aria-label="Open the World War I timeline"
    class="war-service"
    href="/tunnellers/26/wwi-timeline"
  >
    <div>
      <p>
        World War I (1914-1918)
      </p>
      <span>
        New Zealand Tunnellers
      </span>
    </div>
    <div
      class="arrow"
    >
      â†’
    </div>
  </a>
</DocumentFragment>
`;
